modules/internal/localeModels/flat/LocaleModel.chpl

```diff
@@ -395,20 +400,14 @@ module LocaleModel {
     const node = chpl_nodeFromLocaleID(loc);
     if (node == chpl_nodeID) {
       if __primitive("task_get_serial") then
-        // don't call the runtime nb execute_on function if we can stay local
         chpl_ftable_call(fn, args);
       else
-        // We'd like to use a begin, but unfortunately doing so as
-        // follows does not compile for --no-local:
-        // begin chpl_ftable_call(fn, args);
-        chpl_comm_execute_on_nb(node, chpl_sublocFromLocaleID(loc),
-                          fn, args, args_size);
+        chpl_task_taskCallFTable(fn, args, args_size, c_sublocid_any);
```

Comment:
No, I don't believe so.  It would be more clear, but I think it would ultimately be slower because of the machinery we would introduce to pass args to the begin-stmt-body, and this is a place where reducing overhead is important.